Rebranding of code in libvalkeycluster

bjosv opened the pull request:
Presumes that libvalkey (old hiredis) is already renamed as in branch [renaming](https://github.com/valkey-io/libvalkey/tree/renaming).

Part of #11

mkmkme commented on libvalkeycluster/tests/scripts/redis-cluster:
This file is still called `redis-cluster`, is that intentional?

bjosv replied:
Oh, thats a bit weird, will fix

mkmkme commented on libvalkeycluster/valkeycluster.c:
```diff
@@ -197,8 +197,8 @@ static unsigned int keyHashSlot(char *key, int keylen) {
 
-static void __redisClusterSetError(redisClusterContext *cc, int type,
-                                   const char *str) {
+static void __valkeyClusterSetError(valkeyClusterContext *cc, int type,
```
branch `rename` removed the double underscore as function name prefix in commit "Fix ISO C naming violations"

bjosv replied:
Separate PR

bjosv replied:
Covered by issue #19 and PR #21

mkmkme commented on libvalkeycluster/valkeycluster.c:
```diff
@@ -1529,70 +1533,70 @@ void redisClusterFree(redisClusterContext *cc) {
  * When no set of reply functions is given, the default set will be used. */
-static redisClusterContext *_redisClusterConnect(redisClusterContext *cc,
-                                                 const char *addrs) {
+static valkeyClusterContext *_valkeyClusterConnect(valkeyClusterContext *cc,
```
I think single underscore might also be problematic. Maybe `valkeyClusterConnect_impl` / `valkeyClusterConnectImpl`?

bjosv replied:
Separate PR

bjosv replied:
Added issue #19

mkmkme commented on libvalkeycluster/valkeycluster.h:
```diff
@@ -0,0 +1,358 @@
+ * Copyright (c) 2015-2017, Ieshen Zheng <ieshen.zheng at 163 dot com>
+ * Copyright (c) 2020, Nick <heronr1 at gmail dot com>
+ * Copyright (c) 2020-2021, Bjorn Svensson <bjorn.a.svensson at est dot tech>
+ * Copyright (c) 2021, Red Hat
```
Should something be added to Copyright?

bjosv replied:
Separate PR?

mkmkme replied:
Sounds good

mkmkme commented on libvalkeycluster/valkeycluster.h:
```diff
@@ -0,0 +1,358 @@
+#define VALKEYCLUSTER_SLOTS 16384
+
+#define VALKEY_ROLE_NULL 0
+#define VALKEY_ROLE_MASTER 1
```
Can we change the terminology?

bjosv replied:
Sure, the master/slave terminology should be removed

zuiderkwast replied:
Create follow-up issue?

bjosv replied:
Yes, separate PR

mkmkme commented on libvalkeycluster/valkeycluster.h:
```diff
@@ -0,0 +1,358 @@
+int valkeyClusterUpdateSlotmap(valkeyClusterContext *cc);
+
+/* Internal functions */
+valkeyContext *ctx_get_by_node(valkeyClusterContext *cc,
```
Hmmm I wonder if internal functions should really be in a public header. Maybe they could/should be moved to a separate header?

mkmkme commented on libvalkeycluster/vkutil.h:
```diff
@@ -30,26 +30,26 @@
-#define HI_OK 0
-#define HI_ERROR -1
-#define HI_EAGAIN -2
+#define VK_OK 0
```
Here I also got concerned about possible name clash with Vulkan symbols

bjosv replied:
Issue resolved in other comment, lets keep VK

mkmkme reviewed (commented):
Here's my 7 cents on the PR :)

bjosv commented on libvalkeycluster/adlist.c:
```diff
@@ -27,11 +27,11 @@
  */
-#include <hiredis/alloc.h>
 #include <stdlib.h>
+#include <valkey/alloc.h>
```
In hiredis-cluster we have used clang-format, LLVM style, which is the same base used in [valkey](https://github.com/valkey-io/valkey/blob/unstable/src/.clang-format).
So this order is a result of this [include-style](https://llvm.org/docs/CodingStandards.html#include-style).

michael-grunder replied:
I can clang-format #18 if we want to use that style everywhere.

bjosv commented on libvalkeycluster/adapters/ae.h:
```diff
@@ -28,27 +28,27 @@
 
-#ifndef __HIREDIS_CLUSTER_AE_H__
-#define __HIREDIS_CLUSTER_AE_H__
+#ifndef __VALKEYCLUSTER_AE_H__
```
Ok. This PR is mainly a raw rename and my plan was to fix the header guards in a separate PR.
There are a bunch of names/macros that are not following ISO C naming
see:
https://github.com/valkey-io/libvalkey/commit/c19896ab979345a17ba1ac9905ea1de984cf6ea4
https://github.com/valkey-io/libvalkey/commit/4c14ed72231f3e4a6abedceb7a2a54b0e23dfee6

zuiderkwast replied:
These event loop adapters should be merged with the non-cluster ones, right? We only need one copy of them.

bjosv replied:
Added issue #19

bjosv commented on libvalkeycluster/adlist.c:
```diff
@@ -41,7 +41,7 @@
     struct hilist *list;
 
-    if ((list = hi_malloc(sizeof(*list))) == NULL)
+    if ((list = vk_malloc(sizeof(*list))) == NULL)
```
Oh, that's a project I have missed.
hi_malloc is provided by hiredis and the renaming is originally done in the libvalkey code.
Any other suggestions for @michael-grunder

zuiderkwast replied:
Vulkan Memory Allocator seems to use the prefix "vmk" in their functions. I don't think we need to worry about it. It's not in out user-facing API anyway.

mkmkme replied:
@zuiderkwast you're right. And Vulkan itself uses camel case (`vkFunctionName`) in the functions.

However, it still can clash in defines, because Vulkan has `VK_*` for constants. And I'm not sure whether it can become a problem.

zuiderkwast commented on libvalkeycluster/vkutil.h:
```diff
@@ -30,26 +30,26 @@
 
-#ifndef __HIUTIL_H_
-#define __HIUTIL_H_
+#ifndef __VKUTIL_H_
```
Let's move the files to ../src/ in the same PR? Renaming the file multiple times makes it even harder to track the git history.

bjosv replied:
Moved renamed files to src/
Makefiles and so needs to be merged with hiredis/libvalkey..so Ill wait with them so sync with @michael-grunder

zuiderkwast commented on libvalkeycluster/command.c:
```diff
@@ -65,10 +65,10 @@ typedef struct {
 
-/* Populate the table with code in cmddef.h generated from Redis JSON files. */
-static cmddef redis_commands[] = {
+/* Populate the table with code in cmddef.h generated from Valkey JSON files. */
```
... or just `commands`. In many cases we don't need these prefixes, such as static variables like this one.

bjosv replied:
It probably should support open source redis.
The cmddef.h can be re-regenerated by users if they want to include module specific commands,
and it should be possible to include newer Redis releases in the same way.
`server_commands` is probably better here, will change.

bjosv commented on libvalkeycluster/command.c:
```diff
@@ -200,39 +200,15 @@ char *redis_parse_bulk(char *p, char *end, char **str, uint32_t *len) {
- * only supports the Redis unified protocol for requests.
- */
-void redis_parse_cmd(struct cmd *r) {
+void valkey_parse_cmd(struct cmd *r) {
```
You are right, I will keep the part after "Redis >= 1.2 ".

bjosv replied:
Added in fixup commit

bjosv commented on libvalkeycluster/gencommands.py:
```diff
@@ -3,26 +3,26 @@
 #
 # The key specifications part is documented here:
-# https://redis.io/docs/reference/key-specs/
+# https://valkey.io/docs/reference/key-specs/
```
ah.. keep `https://redis.io/docs/reference/key-specs/` here then?

mkmkme replied:
We can keep it for now and create a ticket for `valkey-doc` to have it

zuiderkwast replied:
We will not have the same paths. I have requested to have the URLs look like redis website's URLs used to look like some years ago.

Keeping redis links is not good either. How about "The key specifications part is documented in man page valkey-key-specs(7)"?

bjosv replied:
Updated to existing:
https://valkey.io/docs/topics/key-specs/

bjosv commented on libvalkeycluster/tests/ct_commands.c:
```diff
@@ -202,23 +203,23 @@ void test_eval(redisClusterContext *cc) {
-    reply = (redisReply *)redisClusterCommand(
-        cc, "eval %s 0", "return redis.call('set','foo','bar')");
+    reply = (valkeyReply *)valkeyClusterCommand(
+        cc, "eval %s 0", "return valkey.call('set','foo','bar')");
```
The test is passing when running when using Valkey 7.2.5
Maybe its changed on unstable?

mkmkme replied:
I might also be wrong about it :)

zuiderkwast replied:
We don't have `valkey.call`. Use `server.call` (or `redis.call` which still works for older versions if we care about those).

bjosv replied:
Lets keep redis.call for now since then we can run tests towards redis in CI.
We will have to add some logic here to determine which server the tests run towards.
There is already a function for getting the server version.

bjosv commented on libvalkeycluster/gencommands.py:
```diff
@@ -3,26 +3,26 @@
-# https://github.com/redis/redis-doc/blob/master/commands.json
+# fetches the command metadata from a running Valkey node) or the file
+# commands.json from the valkey-doc repo can be used as input to this script:
+# https://github.com/valkey-io/valkey-doc/blob/master/commands.json
```
Updated

bjosv commented on libvalkeycluster/gencommands.py:
```diff
@@ -3,26 +3,26 @@
+# https://valkey.io/docs/reference/key-specs/
 #
-# The discussion where this JSON format was added in Redis is here:
+# The discussion where this JSON format was added in Valkey is here:
```
Fixed

mkmkme replied:
Forgot to push? Still shows as "was added in Valkey" for me in this file

zuiderkwast commented:
The adapter .h files include

```c
#include "../valkeycluster.h"
#include <valkey/adapters/glib.h>
```

I.e. they're cluster event loop adapters that depend on the corresponding non-cluster adapters. But the file name is just `src/adapters/glib.h` and similar. What's the idea? Merge the files with non-cluster ones or have two separate ones?

bjosv commented:
`What's the idea? Merge the files with non-cluster ones or have two separate ones?

Oh, good finding. I'll need to update the includes after moving stuff to src/.
It would have been easier if libvalkey was moved first..

The plan was to have a single adapter sourcefile for each eventsystem, i.e. basically the version existing in libvalkey but with the added attach function from libvalkeycluster, like `valkeyClusterGlibAttach()` to start with.
This would make current tests work at least.

bjosv commented:
Maybe its simpler in this stage to not rename and move the cluster adapters.
We can just move the function when the libvalkey version in in place..

zuiderkwast commented:
Yes. What matters in the end is how they are used by users.

```c
#include <valkey/cluster-async.h>
#include <valkey/cluster-adapters/glib.h>
```

Or can we make it work without cluster-specific adapters?

```c
#include <valkey/cluster-async.h>
#include <valkey/adapters/glib.h>
```

mkmkme reviewed (approved):
LGTM. I am sure there's a plenty of issues still, but those are worth creating separate tickets

bjosv commented:
> Or can we make it work without cluster-specific adapters?
> 
> ```c
> #include <valkey/cluster-async.h>
> #include <valkey/adapters/glib.h>
> ```

This is the plan, and it seems to work.
Currently I have added the needed attach functions to libvalkey's adapters, but we should probably add that build flag if a user wants to include cluster support of not.

bjosv commented:
Rebased and updated the includes for the moved cluster files.
The tests and examples need to be moved and their includes needs to be corrected too.
I'll merge this and then we can start creating issues.

zuiderkwast commented on libvalkeycluster/examples/src/example_async.c:
```diff
@@ -46,48 +46,48 @@ void disconnectCallback(const redisAsyncContext *ac, int status) {
-    redisClusterAsyncContext *cc =
-        redisClusterAsyncConnect("127.0.0.1:7000", HIRCLUSTER_FLAG_NULL);
+    valkeyClusterAsyncContext *cc =
+        valkeyClusterAsyncConnect("127.0.0.1:7000", VALKEYCLUSTER_FLAG_NULL);
```
it's just `0`? Can we rename this to VALKEYCLUSTER_FLAG_NONE or just use 0? "Null" sticks out to me.

bjosv replied:
I think it would be nice to replace the flags used in the Cluster API with a new valkeyClusterOptions,
like its done in standalone:
```
valkeyContext *valkeyConnectWithOptions(const valkeyOptions *options);
```

Then we could set any option with that API, and possibly keep the simpler one for easy usage.
```
valkeyClusterContext *valkeyClusterConnectWithOptions(const valkeyClusterOptions *options);
valkeyClusterContext *valkeyClusterConnect(const char *addrs);
```
With this change the cluster api would look the same as the standalone.

zuiderkwast reviewed (approved):
You merged the adapters and made tests pass? Incredible!
